WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -116,42 +134,66 @@ class BattleView extends React.Component {
+                                        {/* Enemy MiniCards!
+                                        First card is not visible because is the same as Compact card */}
                                         {[...Array(5)].map(
                                             (e,i) => {
                                                 return (
                                                     <MiniCardView key={`enemyCard-${i}`}
+                                                                  changeCardsOrder={this.changeCardsEnemyOrder}
+                                                                  cardsOrder={this.state.cardsEnemyOrder[i]}
                                                                   setTranslateX={this.state.enemyMiniCardsTranslateX}
                                                                   enemy cardLevel={this.state.cardLevels[i]}
-                                                                  visible={!(i===0)} animationDuration={`0.${10 - i}`}
-                                                                  cardImage={this.state.icons[5 - i]}/>
+                                                                  animationDuration={`0.${9 - i}`}
+                                                                  cardImage={this.state.icons[i]}/>
                                                 );
                                             }
                                         )}
                                     </FlexGapContainer>
                                 </ColumnGapContainer>
-                                <CompactCardView cardImage={icon1} cardName={'Karta 1'}
-                                                 setWidth={'124px'} cardLevel={3} setHeight={'200px'}
-                                                 setTranslateX={this.state.enemyCompactCardTranslateX}
-                                                 setMargin={'0 0 0 10px'} />
+                                {/* Enemy Compact Card! Particular Compact Card is visible if order === 1 */}
+                                {[...Array(5)].map(
+                                    (e,i) => {
+                                        return (
+                                            <CompactCardView key={`enemyCompactCard-${i}`}
+                                                             cardsOrder={this.state.cardsEnemyOrder[i]}
+                                                             cardImage={this.state.icons[i]} cardName={'Karta 1'}
+                                                             setWidth={'124px'} cardLevel={3} setHeight={'200px'}
+                                                             setTranslateX={this.state.enemyCompactCardTranslateX}
+                                                             setMargin={'0 0 0 10px'} />
+                                        );
+                                    }

```

Comment:
O wiele czytelniej by było jakby to wyglądało:

```JavaScript
(e,i) => {
  if(this.state.cardsEnemyOrder[i] === 1)
    return (
        <CompactCardView key={`enemyCompactCard-${i}`}
                         cardsOrder={this.state.cardsEnemyOrder[i]}
                         cardImage={this.state.icons[i]} cardName={'Karta 1'}
                         setWidth={'124px'} cardLevel={3} setHeight={'200px'}
                         setTranslateX={this.state.enemyCompactCardTranslateX}
                         setMargin={'0 0 0 10px'} />
    );
}
```
Tak to się trzeba przekopać do `CompactCardView`, potem znaleźć, że przekazywany jest ten `cardsOrder` do `Div` i tam ogarnąć funkcje `visibleHandler`

Reply:
Ogólnie jest to też duże powtórzenie kodu, na dole jest robione dokładnie to samo. Możnaby zrobić jakąś funkcje, do której podajesz jako parametr `renderEnemyCards` i tam na podstawie tego parametru zmieniasz pomniejsze rzeczy

Reply:
Kurde, tylko jak teraz zrobię to tym ifem, to animacja nie zadziała...

Reply:
W sumie to nie wiem jak to zrobić, aby nie znikał ten komponent instantowo.
Jak kolejność się zmieni, to element nie pasujący do tego ifa zniknie z DOM i nie ma możliwości wtedy animacji opacity.

Reply:
jasne, to nie kombinujmy z tym teraz